### PR TITLE
Use native types

### LINKED ISSUE #170: Add a wrapper to wrap the response when request from LRD UI.
We should add a `data` wrapper to wrap the response in order to inject `_lrd` as an object.

```
{
    "data": (response from controller),
    "_lrd": (LRD responses)
}
```

I will start working on the backend. The frontend should be updated to retrieve details from the new format.

### PR DESCRIPTION
**Pull request by kitloong**
Fix #170, #143, #120

### CODE REVIEW

File: src/Doc.php [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    {
+        $result = [
+            'uri'                  => $this->uri,
+            'middlewares'          => $this->middlewares,
+            'controller'           => $this->controller,
+            'controller_full_path' => $this->controllerFullPath,
+            'method'               => $this->method,
+            'http_method'          => $this->httpMethod,
+            'rules'                => $this->rules,
+            'doc_block'            => $this->docBlock,
```
Comment (kevincobain2000) [inline src/Doc.php:334]: Memo: 
needs to updated on frontend `docBlock -> doc_block`

File: src/LaravelRequestDocs.php [hunk 2]
```diff
@@ -2,261 +2,271 @@
...
-                $requestClass = null;
                 try {
-                    $reflection = new ReflectionClass($requestClassName);
-                    $requestClass = $reflection->newInstanceWithoutConstructor();
-                } catch (Throwable $th) {
-                    //throw $th;
-                }
+                    $requestClassName = $namedType->getName();
+                    $reflectionClass  = new ReflectionClass($requestClassName);
+                    $requestObject    = $reflectionClass->newInstanceWithoutConstructor();
```
Comment (kevincobain2000) [inline src/LaravelRequestDocs.php:235]: code of #120 needs to be added here

File: tests/LaravelRequestDocsMiddlewareTest.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Route;
+use Rakutentech\LaravelRequestDocs\LaravelRequestDocsMiddleware;
+
+class LaravelRequestDocsMiddlewareTest extends TestCase
+{
+    public function testMissingLRDHeader()
+    {
+        Route::get('middleware', function () {
```
Comment (kevincobain2000) [inline tests/LaravelRequestDocsMiddlewareTest.php:14]: shouldn't this also have `data =>`?
Also see #178

Comment (kitloong) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: Should not, this is the case where `X-Request-LRD` header is not exists.
We do not mutate user's response if the request do not come from LRD UI

Comment (kevincobain2000) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: But then the front won’t display the correct results? Since now the front end looks for data.

Comment (kitloong) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: I think this is the scenario, let see if I am missing some info:

```php
# Register middleware
Kernel.php

protected $middlewareGroups = [
    'api' => [
        \Rakutentech\LaravelRequestDocs\LaravelRequestDocsMiddleware::class,
    ];
];
```

1. Request from LRD UI 
   Send request api/v1/health 
   -> LRD UI always add X-Request-LRD header.
   -> Add data wrapper + _lrd details.

2. Request from browser/postman/curl
   Send request api/v1/health
   -> No X-Request-LRD header.
   -> Should not mutate response.

Comment (kevincobain2000) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: Will reply tomorrow. Do not merge.

Comment (kevincobain2000) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: Yes you are right. Good catch. 
but will check one more thing tomorrow.

Comment (kevincobain2000) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: Good catch.
Ok so if `app.debug` is false, the response on LRD will become empty if this is not changed.
Some users may want to use it on prod, just don't want to share SQL, logs and Events.

Screen Shot 2023-03-04 at 11 48 49

How about following?

```
        if (!config('app.debug') && $request->headers->has('X-Request-LRD')) {
            $response    = $next($request);
            $jsonContent = json_encode([
                'data' => $response->getData()
            ]);
            $response->setContent($jsonContent);
            return $response;
        }

        if (!config('app.debug')) {
            return $next($request);
        }
```

Comment (kevincobain2000) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: Pushed for re-review.
**Critical**

Comment (kevincobain2000) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: Probably we should provide a `config('request-docs.enabled')` as well for security reasons

Comment (kitloong) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: Thank you.

Yes, adding `config('request-docs.enabled')` is better.

Comment (kevincobain2000) [reply tests/LaravelRequestDocsMiddlewareTest.php:14]: added

**Pull request merged**